src/m365/spo/commands/user/user-ensure.ts

```diff
@@ -0,0 +1,113 @@
+import { Logger } from '../../../../cli/Logger';
+import GlobalOptions from '../../../../GlobalOptions';
+import request, { CliRequestOptions } from '../../../../request';
+import { aadUser } from '../../../../utils/aadUser';
+import { validation } from '../../../../utils/validation';
+import SpoCommand from '../../../base/SpoCommand';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+export interface Options extends GlobalOptions {
+  webUrl: string;
+  aadId?: string;
+  userName?: string;
+}
+
+class SpoUserEnsureCommand extends SpoCommand {
+  public get name(): string {
+    return commands.USER_ENSURE;
+  }
+
+  public get description(): string {
+    return 'Ensures that a user is available on a specific site';
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+    this.#initOptionSets();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        aadId: typeof args.options.aadId !== 'undefined',
+        userName: typeof args.options.userName !== 'undefined'
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-u, --webUrl <webUrl>'
+      },
+      {
+        option: '--aadId [aadId]'
+      },
+      {
+        option: '--userName [userName]'
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        const isValidSharePointUrl: boolean | string = validation.isValidSharePointUrl(args.options.webUrl);
+        if (isValidSharePointUrl !== true) {
+          return isValidSharePointUrl;
+        }
+
+        if (args.options.aadId && !validation.isValidGuid(args.options.aadId)) {
+          return `${args.options.aadId} is not a valid GUID.`;
+        }
+
+        if (args.options.userName && !validation.isValidUserPrincipalName(args.options.userName)) {
+          return `${args.options.userName} is not a valid userName.`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push({ options: ['aadId', 'userName'] });
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    if (this.verbose) {
+      logger.logToStderr(`Ensuring user ${args.options.aadId || args.options.userName} at site ${args.options.webUrl}`);
+    }
+
+    try {
+      const requestBody = {
+        logonName: args.options.userName || await aadUser.getUpnByUserId(args.options.aadId!)
```

Comment:
This is a nice one-liner. But as it stands we're now missing out on a verbose line like 'Retrieving upn for user'.
Could you refactor it so that it includes a verbose line in this scenario?